src/main/java/org/opensearch/security/privileges/ActionPrivileges.java

```diff
@@ -1026,7 +1026,23 @@ static class StatefulIndexPrivileges {
                                     if (indicesEntry.getValue() instanceof IndexAbstraction.Alias) {
                                         // For aliases we additionally add the sub-indices to the privilege map
                                         for (IndexMetadata subIndex : indicesEntry.getValue().getIndices()) {
-                                            indexToRoles.get(subIndex.getIndex().getName()).add(roleName);
+                                            String subIndexName = subIndex.getIndex().getName();
+                                            // We need to check whether the subIndex is part of the global indices
+                                            // metadata map because that map has been filtered by relevantOnly().
+                                            // This method removes all closed indices and data stream backing indices
+                                            // because these indices get a separate treatment. However, these indices
+                                            // might still appear as member indices of aliases. Trying to add these
+                                            // to the SubSetBuilder indexToRoles would result in an IllegalArgumentException
+                                            // because the subIndex will not be part of the super set.
+                                            if (indices.containsKey(subIndexName)) {
+                                                indexToRoles.get(subIndexName).add(roleName);
+                                            } else {
+                                                log.debug(
+                                                    "Ignoring member index {} of alias {}. This is usually the case because the index is closed or a data stream backing index.",
+                                                    subIndexName,
```

Comment:
Thanks for the quick work @nibix ! Should we open a corresponding GH issue to support data streams with optimized evaluation?

Reply:
It is already supported; it is just that the code in question is not responsible for supporting it.

Reply:
When designing the feature, we had the challenge that the data structures must not grow too big. Especially data streams, however, have the habit of creating many backing indices under the hood. For that reason, the code contains a special case for backing indices. Backing indices are not added as member to the data structures. Rather, only an entry for the data stream itself is created. As backing indices have a back reference to the data stream, it is easy to deduct the privileges for the backing indices from the data stream.

Reply:
Thanks as always @nibix !